specs/_features/eip7594/das-core.md

```diff
@@ -0,0 +1,225 @@
+    line_index = node_id % NUMBER_OF_COLUMNS
+    return [LineIndex(all_items[(line_index + i) % len(all_items)]) for i in range(custody_size)]
+```
+
+#### `compute_extended_data`
+
+```python
+def compute_extended_data(data: Sequence[BLSFieldElement]) -> Sequence[BLSFieldElement]:
+    # TODO
+    # pylint: disable=unused-argument
+    ...
+```
+
+#### `compute_extended_matrix`
+
+```python
+def compute_extended_matrix(blobs: Sequence[Blob]) -> FlatExtendedMatrix:
+    matrix = [compute_extended_data(blob) for blob in blobs]
+    return FlatExtendedMatrix(matrix)
+```
+
+#### `get_data_column_sidecars`
+
+```python
+def get_data_column_sidecars(signed_block: SignedBeaconBlock,
+                             blobs: Sequence[Blob]) -> Sequence[DataColumnSidecar]:
+    signed_block_header = compute_signed_block_header(signed_block)
+    block = signed_block.message
+    kzg_commitments_inclusion_proof = compute_merkle_proof(
+        block.body,
+        get_generalized_index(BeaconBlockBody, 'blob_kzg_commitments'),
+    )
+    cells_and_proofs = [compute_cells_and_proofs(blob) for blob in blobs]
+    blob_count = len(blobs)
+    cells = [cells_and_proofs[i][0] for i in range(blob_count)]
+    proofs = [cells_and_proofs[i][1] for i in range(blob_count)]
+    sidecars = []
+    for column_index in range(NUMBER_OF_COLUMNS):
+        column = DataColumn([cells[row_index][column_index]
+                             for row_index in range(blob_count)])
+        kzg_proof_of_column = [proofs[row_index][column_index]
+                               for row_index in range(blob_count)]
+        sidecars.append(DataColumnSidecar(
+            index=column_index,
+            column=column,
+            kzg_commitments=block.body.blob_kzg_commitments,
+            kzg_proofs=kzg_proof_of_column,
+            signed_block_header=signed_block_header,
+            kzg_commitments_inclusion_proof=kzg_commitments_inclusion_proof,
+        ))
+    return sidecars
+```
+
+## Custody
+
+### Custody requirement
+
+Each node downloads and custodies a minimum of `CUSTODY_REQUIREMENT` columns per slot. The particular columns that the node is required to custody are selected pseudo-randomly (more on this below).
+
+A node *may* choose to custody and serve more than the minimum honesty requirement. Such a node explicitly advertises a number greater than `CUSTODY_REQUIREMENT`  via the peer discovery mechanism -- for example, in their ENR (e.g. `custody_lines: 8` if the node custodies `8` columns each slot) -- up to a `NUMBER_OF_COLUMNS` (i.e. a super-full node).
+
+A node stores the custodied columns for the duration of the pruning period and responds to peer requests for samples on those columns.
+
+### Public, deterministic selection
+
+The particular columns that a node custodies are selected pseudo-randomly as a function (`get_custody_lines`) of the node-id and custody size -- importantly this function can be run by any party as the inputs are all public.
```

Comment:
```suggestion
The particular set of consecutive columns that a node custodies is selected pseudo-randomly as a function (`get_custody_lines`) of the node-id and custody size -- importantly this function can be run by any party as the inputs are all public.
```

Reply:
Resolved by changing the custody assignment calculation (https://github.com/ethereum/consensus-specs/pull/3574#issuecomment-1914298172).